src/raytracer/multipath.m

```diff
@@ -123,116 +107,85 @@
         % a single row of ArrayOfPlanes,ArrayOfPoints is fed to
         % singleMultipathGenerator function to know whether a path exists. If a
         % path exists then what are vectors that form the path (stored in
-        % multipath parameter)
+        % multipath parameter)                
+        [isMpc,~,dod,doa,multipath,distance,dopplerFactor,...
+           ~] = singleMultipathGenerator...
+            (iterateNumberOfRowsArraysOfPlanes,orderOfReflection,indexOrderOfReflection,ArrayOfPlanes,...
+            ArrayOfPoints,Reflected,Rx,Tx,CADOutput,...
+            multipath,indexMultipath,velocityTx,velocityRx);
         
-        PolarizationSwitchTemporary = 1;
-        if PolarizationSwitch == 1 && switchMaterial == 1
-            for orderOfReflectionTemporary = 1:orderOfReflection
-                dielectricConstant = MaterialLibrary.DielectricConstant(...
-                    arrayOfMaterials(indexMultipath,orderOfReflectionTemporary));
-                if dielectricConstant ~= 0
-                    nt_array(orderOfReflectionTemporary) = dielectricConstant;
-                else
-                    nt_array = [];
-                    PolarizationSwitchTemporary = 0;
-                    break
-                end
-            end
+        % Apply node rotation
+        dod = coordinateRotation(dod,[0 0 0], qTx.angle, 'frame');
+        doa = coordinateRotation(doa,[0 0 0], qRx.angle, 'frame');
+        
+        % Compute reflection loss
+        if  switchMaterial == 1
+            reflectionLoss = getReflectionLoss(MaterialLibrary,...
+                arrayOfMaterials(indexMultipath,:), 'randOn', qdGeneratorSwitch);
         else
-            nt_array = [];
-            PolarizationSwitchTemporary = 0;
+            % Assumption: 10dB loss at each reflection
+            reflectionLoss = 10*orderOfReflection; 
```

Comment:
Yep it is explained in the documentation (or it will be explained soon). If the material library is found, the reflection loss is computed from the material library, else we assume 10dB loss per reflection. No specific reason for the value, we found it a reasonable value

Reply:
I'm not sure about this point: what if one wants to setup different losses per reflection/material?
Maybe it would be better to keep no reflection loss by default, document it, and leave the possibility to the user to choose a custom reflection loss depending on his purpose.
Another possibility would be to add a custom reflection loss value to the config file, maybe setting the default to 10 dB and properly documenting it.
What do you think about this?

Reply:
Good suggestion, I'll add reflection loss as input